src/model/price-range-impl.ts

```diff
@@ -2,6 +2,24 @@ import { isNumber } from '../helpers/strict-type-checks';
 
 import { PriceRange } from './series-options';
 
+function ensureFiniteWithFallback(value: number, fallback: number): number {
+	return Number.isFinite(value) ? value : fallback;
+}
+
+function computeFiniteResult(
+	method: (...values: number[]) => number,
+	valueOne: number,
+	valueTwo: number,
+	fallback: number
+): number {
+	const firstFinite = Number.isFinite(valueOne);
+	return firstFinite && Number.isFinite(valueTwo)
+		? method(valueOne, valueTwo)
+		: firstFinite
+		? valueOne
+		: ensureFiniteWithFallback(valueTwo, fallback);
+}
```

Comment:
```suggestion
function computeFiniteResult(
	method: (...values: number[]) => number,
	valueOne: number,
	valueTwo: number,
	fallback: number
): number {
	return method(
		ensureFiniteWithFallback(valueOne, fallback),
		ensureFiniteWithFallback(valueTwo, fallback)
	);
}
```

Reply:
We would like the fallback to be possibly an Infinite value so this wouldn't work.

We expect that this: `computeFiniteResult(Math.min, -Infinity, 55, -Infinity)` should return `55`. With this change it would return `-Infinity`.

Essentially the fallback should be used if BOTH values are Infinite.

Where we are using the code, we would like it to be Infinite if both values are infinite, but finite if either value (or both) are finite.

I suspect that the function (and the variables) could be named better.

Maybe: `fallback` → `fallbackIfBothNonFinite`

Reply:
Then probably we need to filter out infinite values first

```ts
function finiteValuesOrNull(...values: number[]): numer[] | null {
  const result = values.filter(Numer.isFinite);
  return result.length ? result : null;
}
...
return new PriceRangeImpl(
  Math.min(...(finiteValuesOrNull(this.minValue(), anotherRange.minValue()) || [-Infinity])),
  Math.max(...(finiteValuesOrNull(this.maxValue(), anotherRange.maxValue()) || [Infinity]))
);
```

Reply:
My original commit used `filter` but I changed it to the current version because I wanted to avoid using arrays so that the function is a bit faster (avoid creating and spreading array if possible).

The original commit had:
```typescript
function computeFiniteResult(
 	values: number[],
	method: (...values: number[]) => number,
 	fallback: number
 ): number {
 	const result = method(...values.filter(Number.isFinite));
 	return Number.isFinite(result) ? result : fallback;
 }
```

This function will be called on every scroll / scale / etc... so I tried to make it a bit faster (didn't actually benchmark the difference).

Reply:
OK, anyway I think we should get rid of `ensureFiniteWithFallback`

```suggestion
function computeFiniteResult(
	method: (...values: number[]) => number,
	valueOne: number,
	valueTwo: number,
	fallback: number
): number {
	const firstFinite = Number.isFinite(valueOne);
	const secondFinite = Number.isFinite(valueTwo);
	
	if (firstFinite && secondFinite) {
		return method(valueOne, valueTwo);
	}
	
	return !firstFinite && !secondFinite ? fallback : (firstFinite ? valueOne : valueTwo);
}
```